File: Api/gestiMed2024/src/java/Clases/Medico.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Clases;

import Listas.ListaPaciente;
import java.io.Serializable;
import java.util.List;

/**
 *
 * @author joans
 */
public class Medico implements Serializable{
    
    private String dni;
    private String nombre;
    private String apellidos;
    private String email;
    private String contrasenya;
    private String numContacto;
    private ListaPaciente pacienteList;
    
    public Medico() {
    }

    public Medico(String dni, String nombre, String apellidos, String email, String contrasenya, String numContacto) {
        this.dni = dni;
        this.nombre = nombre;
        this.apellidos = apellidos;
        this.email = email;
        this.contrasenya = contrasenya;
        this.numContacto = numContacto;
    }

    public String getDni() {
        return dni;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidos() {
        return apellidos;
    }

    public void setApellidos(String apellidos) {
        this.apellidos = apellidos;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getContrasenya() {
        return contrasenya;
    }

    public void setContrasenya(String contrasenya) {
        this.contrasenya = contrasenya;
    }

    public String getNumContacto() {
        return numContacto;
    }

    public void setNumContacto(String numContacto) {
        this.numContacto = numContacto;
    }

    public ListaPaciente getPacienteList() {
        return pacienteList;
    }

    public void setPacienteList(ListaPaciente pacienteList) {
        this.pacienteList = pacienteList;
    }

    @Override
    public String toString() {
        return "Medico{" + "dni=" + dni + ", nombre=" + nombre + ", apellidos=" + apellidos + ", email=" + email + ", contrasenya=" + contrasenya + ", numContacto=" + numContacto + ", pacienteList=" + pacienteList + '}';
    }
    
    
}

File: WPF/gestiMed2024/gestiMed2024/clases/Medico.cs
using gestiMed2024.listas;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace gestiMed2024.clases
{
    class Medico
    {
        private string dni;
        private string nombre;
        private string apellidos;
        private string email;
        private string contrasenya;
        private string numContacto;
        private ListaPaciente pacientes;

        //Constructores 
        public Medico() { }
        public Medico(string apellidos,string contrasenya, string dni, string email, string nombre, string numContacto, ListaPaciente pacientes)
        {
            this.apellidos = apellidos;
            this.contrasenya = contrasenya;
            this.dni = dni;
            this.email = email;
            this.nombre = nombre;
            this.numContacto = numContacto;
            this.pacientes = pacientes;
        }
        public Medico(string apellidos, string contrasenya, string dni, string email, string nombre, string numContacto)
        {
            this.apellidos = apellidos;
            this.contrasenya = contrasenya;
            this.dni = dni;
            this.email = email;
            this.nombre = nombre;
            this.numContacto = numContacto;
        }
        //Getters
        public string getDni() { return dni; }
        public string getNombre() { return nombre; }
        public string getApellidos() { return apellidos; }
        public string getEmail() { return email; }
        public string getContrasenya() { return contrasenya; }
        public string getNumContacto() { return numContacto; }
        public ListaPaciente getPacientes() { return pacientes; }
        // Setters
        public void setDni(string dni) { this.dni = dni; }
        public void setNombre(string nombre) { this.nombre = nombre; }
        public void setApellidos(string apellidos) { this.apellidos = apellidos; }
        public void setEmail(string email) { this.email = email; }
        public void setContrasenya(string contrasenya) { this.contrasenya = contrasenya; }
        public void setPacientes(ListaPaciente pacientes) { this.pacientes = pacientes; }
    }
}
